### PR TITLE
Tidy reliable sessions topics

### PR DESCRIPTION
**Pull request by guardrex**
On my way to getting a review done, I noticed some layout and style problems with the reliable sessions group of topics, so I made a quick pass on this group for guidelines adherence.

### CODE REVIEW

**Comment by GuardRex** [comment]
@mairaw Something's up with the checks, right? ... a known issue?

**Comment by GuardRex** [comment]
Ready for :eyes: now.

File: docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md [hunk 1]
```diff
@@ -15,67 +15,75 @@ author: "Erikre"
...
- Where "XX" is the number of connections needed. The minimum in this case should be 4.
\ No newline at end of file
+
+This topic discusses best practices for reliable sessions.
+
+## Setting MaxTransferWindowSize
+
+Reliable sessions in [!INCLUDE[indigo1](../../../../includes/indigo1-md.md)] use a transfer window to hold messages on the client and service. The configurable property <xref:System.ServiceModel.Channels.ReliableSessionBindingElement.MaxTransferWindowSize%2A> indicates how many messages the transfer window can hold.
+
+On the sender, this indicates how many messages the transfer window can hold while waiting for acknowledgements; on the receiver it indicates how many messages to buffer for the service.
```
Comment (rpetrusha) [inline docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md]: comma after "receiver"

File: docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md [hunk 2]
```diff
@@ -15,67 +15,75 @@ author: "Erikre"
...
+
+Choosing the right size impacts the efficiency of the network and the optimal capacity of the service. The following sections detail what to consider when choosing a value for this property and the impact of the value.
+
+The default transfer window size is eight messages.
+
+### Efficient use of the network
+
+In this context, the term *network* corresponds to everything between a client (sender) and a service (receiver) used as the basis of communication. This includes the transport connections and any intermediary or bridges in-between, including SOAP routers or HTTP proxies/firewalls.
```
Comment (rpetrusha) [inline docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md]: everything used as the basis of communication between a client (sender) and a service (receiver).
in-between --> in between

File: docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md [hunk 3]
```diff
@@ -15,67 +15,75 @@ author: "Erikre"
...
+
+On the sender, the property <xref:System.ServiceModel.Channels.ReliableSessionBindingElement.MaxTransferWindowSize%2A> indicates how many messages its transfer window can hold while waiting for acknowledgements. If the network latency is high and in order to ensure a responsive sender and effective network utilization, you should increase the transfer window size.
+
+For example even if the sender keeps up with data rate, latency could be high if several intermediaries exist between the sender and receiver or the data must pass through a lossy intermediary or network. Thus, the sender has to wait for acknowledgements for the messages in its transfer window before accepting new messages to send on the wire. The smaller the buffer with high latency, the less effectively the network utilization. On the other hand, too high a transfer window size may impact the service because the service may need to catch up to the high rate of data sent by the client.
```
Comment (rpetrusha) [inline docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md]: less effectively --> less effective

File: docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md [hunk 4]
```diff
@@ -15,67 +15,75 @@ author: "Erikre"
...
+
+### Running the service to capacity
+
+As much as the network is used efficiently, ideally you also want the service to run at optimal capacity. The transfer window size property on the receiver indicates how many messages the receiver can buffer. This message buffering helps not only the network flow control but also enables the service to run to full capacity. For example if the buffer is one message and messages arrive faster than the service can process them, then the network might drop messages and capacity might be wasted or underutilized.
+
+Using a buffer increases the availability of the service as it concurrently receives and buffers a message while processing the previously received messages.
+
+It's recommended that you use the same `MaxTransferWindowSize` on both the sender and receiver.
```
Comment (rpetrusha) [inline docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md]: It's recommended --> We recommend

File: docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md [hunk 5]
```diff
@@ -15,67 +15,75 @@ author: "Erikre"
...
+
+### Enabling flow control
+
+*Flow control* is a mechanism that ensures that the sender and receiver keep pace with each other, that is, the messages are consumed and acted upon as fast as they're produced. The transfer window size on the client and service ensures that the sender and receiver are within a reasonable window of synchronization.
+
+It's highly recommended that you set the property <xref:System.ServiceModel.Channels.ReliableSessionBindingElement.FlowControlEnabled%2A> to `true` when you're using a reliable session between a [!INCLUDE[indigo2](../../../../includes/indigo2-md.md)] client and a [!INCLUDE[indigo2](../../../../includes/indigo2-md.md)] service.
```
Comment (rpetrusha) [inline docs/framework/wcf/feature-details/best-practices-for-reliable-sessions.md]: It's highly recommended --> We highly recommend

File: docs/framework/wcf/feature-details/how-to-secure-messages-within-reliable-sessions.md [hunk 6]
```diff
@@ -15,52 +15,54 @@ author: "BrucePerlerMS"
...
-    <wsHttpBinding>  
-    <binding name="MessageSecurity">  
-        <security mode="Message" />  
-           <message clientCredentialType = " Certificate" />  
-        </security>  
-    </binding>  
-    </wsHttpBinding >  
-    ```
\ No newline at end of file
+> When using message security with reliable sessions and if the client is not authenticated, Reliable Messaging attempts to authenticate the client until a timeout occurs instead of throwing an exception upon first failure.
```
Comment (rpetrusha) [inline docs/framework/wcf/feature-details/how-to-secure-messages-within-reliable-sessions.md]: instead of "and", I'd use a comma, or rephrase this to "When using message security with reliable sessions, Reliable Messaging attempts to authenticate an unauthenticated client..."

**Review by rpetrusha — APPROVED** [review]
There are a couple of suggestions to clear up some awkward sentences, but otherwise LGTM.
